dashboard/app/lib/javascripts/dashboard/main.js

```diff
@@ -65,6 +60,90 @@ window.Dashboard = {
 		Marbles.history.navigate('login'+ redirectPath);
 	},
 
+	__catchInsecurePingResponse: function(httpsArgs) {
+		var httpsXhr = httpsArgs[1], self = this,
+			handleSuccess, handleError;
+
+		handleSuccess = function (httpArgs) {
+			var httpXhr = httpArgs[1];
+			// https did not work but http did...something is wrong with the cert
+			self.Dispatcher.handleAppEvent({
+				name: "HTTPS_CERT_MISSING",
+				status: httpXhr.status
+			});
+		};
+		handleError = function (httpArgs) {
+			var httpXhr = httpArgs[1];
+			if (httpXhr.status === 0) {
+				// https is failing as well...service is unavailable
+				self.Dispatcher.handleAppEvent({
+					name: "SERVICE_UNAVAILABLE",
+					status: httpXhr.status
+				});
+			}
+			// https did not work but http did without a network error
+			// => missing ssl exception for controller
+			self.Dispatcher.handleAppEvent({
+				name: "HTTPS_CERT_MISSING",
+				status: httpXhr.status
+			});
+		};
+
+		if (httpsXhr.status === 0) {
+			// https is unavailable, let's see if http works
+			self.client.ping("controller", "http").catch(handleError).then(handleSuccess);
+			return;
+		}
+		// We got something else than 0 and it's an error. This results in SERVICE_UNAVAILABLE
+		self.Dispatcher.handleAppEvent({
+			name: "SERVICE_UNAVAILABLE",
+			status: httpsXhr.status
+		});
+	},
+
+	__catchSecurePingResponse: function(args) {
+		var xhr = args[1];
+		if (xhr.status === 0) {
+			// We were not able to access the controller due to a network error (ssl, timeout)
+			// In order to understand what's happening, we have to switch to http.
+			this.Dispatcher.handleAppEvent({
+				name: "CONTROLLER_UNREACHABLE_FROM_HTTPS",
+				status: xhr.status
+			});
+			return;
+		}
+		// We got something else than 0 and it's an error. This results in SERVICE_UNAVAILABLE
+		this.Dispatcher.handleAppEvent({
+			name: "SERVICE_UNAVAILABLE",
+			status: xhr.status
+		});
+	},
+
+	__isCertInstalled: function() {
+		var self = this;
+		if (window.location.protocol === "https:") {
+			this.client.ping("controller", "https").catch(this.__catchSecurePingResponse.bind(this));
+		} else {
+			this.client.ping("controller", "https")
+				.catch(this.__catchInsecurePingResponse.bind(this))
+				.then(function(args) {
+					var xhr = args[1];
+					if (xhr.status !== 200) {
+						// If we don't make sure that the response was successful, we might end up in an
+						// infinite loop.
+						return;
+					}
+					// TODO this probably should be Dashboard.config.API_SERVER
+					self.client.ping("dashboard", "https").then(function() {
```

Comment:
Is this needed? I think it's safe to assume we can reach the dashboard over https if the controller is reachable.